code/modules/mob/living/carbon/superior_animal/superior_animal.dm

```diff
@@ -533,20 +533,20 @@
 	if (can_burrow && bad_environment)
 		evacuate()
 
-	if(!AI_inactive)
+	if(!AI_inactive && !weakened) //we dont need to handle ai if we're disabled
```

Comment:
```suggestion
	if(!AI_inactive || !weakened) //we dont need to handle ai if we're disabled
```
Otherwise we are active even if are AI is off

Reply:
no